plugins/builds/artifacts/getAll.js

```diff
@@ -0,0 +1,102 @@
+'use strict';
+
+const boom = require('@hapi/boom');
+const joi = require('joi');
+const jwt = require('jsonwebtoken');
+const logger = require('screwdriver-logger');
+const request = require('got');
+const schema = require('screwdriver-data-schema');
+const { v4: uuidv4 } = require('uuid');
+const idSchema = schema.models.build.base.extract('id');
+const AdmZip = require('adm-zip');
+
+module.exports = config => ({
+    method: 'GET',
+    path: '/builds/{id}/artifacts',
+    options: {
+        description: 'Get a zipped file including all build artifacts',
+        notes: 'Redirects to store with proper token',
+        tags: ['api', 'builds', 'artifacts'],
+        auth: {
+            strategies: ['session', 'token'],
+            scope: ['user', 'build', 'pipeline']
+        },
+
+        handler: async (req, h) => {
+            const { name: artifact, id: buildId } = req.params;
+            const { credentials } = req.auth;
+            const { canAccessPipeline } = req.server.plugins.pipelines;
+            const { buildFactory, eventFactory } = req.server.app;
+
+            return buildFactory.get(buildId)
+                .then(build => {
+                    if (!build) {
+                        throw boom.notFound('Build does not exist');
+                    }
+
+                    return eventFactory.get(build.eventId);
+                })
+                .then(event => {
+                    if (!event) {
+                        throw boom.notFound('Event does not exist');
+                    }
+
+                    return canAccessPipeline(credentials, event.pipelineId, 'pull', req.server.app);
+                })
+                .then(async () => {
+                    const token = jwt.sign({
+                        buildId, artifact, scope: ['user']
+                    }, config.authConfig.jwtPrivateKey, {
+                        algorithm: 'RS256',
+                        expiresIn: '5s',
```

Comment:
the expiry of 5 sec may be too short when downloading large files, consider changing to may be 10m